pkg/cluster/deploy.go

```diff
@@ -30,3 +45,133 @@ func (u *simpleUpgrader) Deploy(ctx context.Context, cs *api.OpenShiftManagedClu
 	}
 	return nil
 }
+
+type scalesetName string
+type instanceName string
+type hash string
+
+type vmInfo struct {
+	ComputerName instanceName `json:"computerName,omitempty"`
+	ScalesetHash hash         `json:"scalesetHash,omitempty"`
+}
+
+func hashScaleSets(azuretemplate map[string]interface{}) (map[scalesetName]hash, error) {
+	ssHashes := make(map[scalesetName]hash)
+	for _, r := range jsonpath.MustCompile("$.resources[?(@.type='Microsoft.Compute/virtualMachineScaleSets')]").Get(azuretemplate) {
+		original, ok := r.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		// deep-copy the ARM template since we are mutating it below.
+		resource := deepCopy(original)
+
+		// cleanup capacity so that no unnecessary VM rotations are going
+		// to occur because of a scale up/down.
+		jsonpath.MustCompile("$.sku.capacity").Delete(resource)
+
+		// filter out the nsg dependsOn entry since we remove it
```

Comment:
I think you can safely just `jsonpath.MustCompile("$.dependsOn").Delete(resource)` here.  Much easier.

Reply:
Isn't this going to remove all dependsOn entries? There are more in the scale sets than just the nsg dep.

Reply:
I guess we can assume that a new dependency will also mean a change somewhere else in the scaleset so maybe dropping all the dependsOn entries is an ok thing to do here.